apps/heft/src/stages/CleanStage.ts

```diff
@@ -36,16 +37,43 @@ export interface ICleanStageOptions {
  */
 export interface ICleanStageContext extends IStageContext<CleanStageHooks, ICleanStageProperties> {}
 
+interface ICleanConfigurationJson {
+  pathsToDelete: string[];
+}
+
 export class CleanStage extends StageBase<CleanStageHooks, ICleanStageProperties, ICleanStageOptions> {
   public constructor(heftConfiguration: HeftConfiguration, loggingManager: LoggingManager) {
     super(heftConfiguration, loggingManager, CleanStageHooks);
   }
 
-  protected getDefaultStageProperties(options: ICleanStageOptions): ICleanStageProperties {
+  protected async getDefaultStagePropertiesAsync(
+    options: ICleanStageOptions
+  ): Promise<ICleanStageProperties> {
+    const cleanConfigurationFileLoader: ConfigurationFile<ICleanConfigurationJson> = new ConfigurationFile<
+      ICleanConfigurationJson
+    >(path.resolve(__dirname, '..', 'schemas', 'clean.schema.json'), {
```

Comment:
I also still want to introduce the ".defaults.json" concept from PR https://github.com/microsoft/rushstack/pull/2054

Should we convert the `schemaPath` positional parameter to be part of the options, like below?
```ts
new ConfigurationFile<
        IPluginConfigurationJson
      >({
        jsonSchemaPath: path.join(__dirname, '..', 'schemas', 'clean.schema.json'), // <----
        jsonDefaultsPath: path.join(__dirname, '..', 'schemas', 'clean.defaults.json'), // <----
        propertyInheritanceTypes: { plugins: InheritanceType.append },
        jsonPathMetadata: {
          '$.plugins.*.plugin': {
            pathResolutionMethod: PathResolutionMethod.NodeResolve
          }
        }
      }
```

That way the strings won't get confused later when we introduce `jsonDefaultsPath` alongside `jsonSchemaPath`.

Reply:
Yeah I'm cool with that.